api/desecapi/views.py

```diff
@@ -41,6 +46,52 @@ def get_client_ip(request):
     return request.META.get('REMOTE_ADDR')
 
 
+class TokenCreateView(djoser.views.TokenCreateView):
+
+    def _action(self, serializer):
+        user = serializer.user
+        token = Token(user=user)
+        token.save()
+        user_logged_in.send(sender=user.__class__, request=self.request, user=user)
+        token_serializer_class = DjoserTokenSerializer
+        return Response(
+            data=token_serializer_class(token).data,
+            status=status.HTTP_201_CREATED,
+        )
+
+
+class TokenDestroyView(djoser.views.TokenDestroyView):
+
+    def post(self, request):
+        _, token = auth.TokenAuthentication().authenticate(request)
+        token.delete()
+        user_logged_out.send(
+            sender=request.user.__class__, request=request, user=request.user
+        )
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+class TokenViewSet(mixins.CreateModelMixin,
+                   mixins.DestroyModelMixin,
```

Comment:
I am not sure how exactly this works -- I thought that create and list operations are for list endpoints while the destroy action aims at a specific model instance. Where is that taken care of? I suppose it's some magic interaction with the `SimpleRouter` used in `urls.py`. Can you elaborate briefly?